edx_sga/tasks.py

```diff
@@ -0,0 +1,309 @@
+                    loc=location,
+                    sha1=answer['sha1'],
+                    ext=os.path.splitext(answer['filename'])[1]
+                )
+
+                destination_file = os.path.join(
+                    destination_path,
+                    "{student_name}_{sha1}{ext}".format(
+                        student_name=student.username,
+                        sha1=answer['sha1'],
+                        ext=os.path.splitext(answer['filename'])[1]
+                    )
+                )
+                try:
+                    file_descriptor = default_storage.open(source_file)
+                    with open(destination_file, 'wb') as file_pointer:
+                        while True:
+                            data = file_descriptor.read(BLOCK_SIZE)
+                            if data == '':  # end of file reached
+                                break
+                            file_pointer.write(data)
+
+                except IOError:
+                    log.exception("Unable to download submission of student=%s", student.username)
+        else:
+            continue
+
+
+def _compress_folder(destination_path):
+    """
+    compress given folder/file
+
+    Args:
+        destination_path (str): path (including name) of folder/file which we want to compress.
+    """
+    make_archive(
+        destination_path,
+        'zip',
+        destination_path
+    )
+
+
+@CELERY_APP.task
+def zip_student_submissions(course_id, block_id, location, user_name):
+    """
+    Task to download all submissions as zip file
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user_name (str): staff user name
+    """
+    submissions_dir_name = get_submissions_dir_name(user_name, block_id, course_id)
+    destination_path = _get_dir_path(submissions_dir_name, location)
+    _mk_dir(destination_path)
+    remove_zip_file(user_name, block_id, course_id)
+    _compress_student_submissions(block_id, course_id, location, destination_path)
+    _compress_folder(destination_path)
+
+
+def get_submissions_dir_name(user_name, block_id, course_id):
+    """
+    returns submission folder name where we are saving files temporary.
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        user_name (unicode): staff user name
+    """
+    return "{username}_submissions_{id}_{course_key}".format(
+        username=user_name,
+        id=hashlib.md5(block_id).hexdigest(),
+        course_key=course_id
+    )
+
+
+def _get_dir_path(submissions_dir_name, location=None):
+    """
+    returns directory path where we are saving files temporary.
+
+    Args:
+        submissions_dir_name (str): name of folder
+        location (Location): location of sga module
+    """
```

Comment:
1. A better name for this would be `get_collected_submissions_dir_path` or something like that
1. Why does this function start with an underscore and `get_submissions_dir_name` doesn't? They are doing very similar things for very similar purposes. I would say they should have the same level of protection
1. In keeping with the function name change I suggested above, `submissions_dir_name` should be changed to something like `submissions_base_name`